lib/new_relic/harvest/telemetry_sdk/dimensional_metrics/harvester.ex

```diff
@@ -0,0 +1,174 @@
+defmodule NewRelic.Harvest.TelemetrySdk.DimensionalMetrics.Harvester do
+  use GenServer
+
+  @moduledoc false
+
+  alias NewRelic.Harvest
+  alias NewRelic.Harvest.TelemetrySdk
+
+  @interval_ms TelemetrySdk.Config.lookup(:dimensional_metrics_harvest_cycle)
+
+  @valid_types [:count, :gauge, :summary]
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [])
+  end
+
+  def init(_) do
+    {:ok,
+     %{
+       start_time: get_start_time(),
+       metrics: %{}
+     }}
+  end
+
+  # API
+
+  @spec report_dimensional_metric(:count | :gauge | :summary, atom() | binary(), any, map()) ::
+          :ok
+  def report_dimensional_metric(type, name, value, attributes) when type in @valid_types do
+    TelemetrySdk.DimensionalMetrics.HarvestCycle
+    |> Harvest.HarvestCycle.current_harvester()
+    |> GenServer.cast({:report, %{type: type, name: name, value: value, attributes: attributes}})
+  end
+
+  def gather_harvest,
+    do:
+      TelemetrySdk.DimensionalMetrics.HarvestCycle
+      |> Harvest.HarvestCycle.current_harvester()
+      |> GenServer.call(:gather_harvest)
+
+  # do not accept more report messages when harvest has already been reported
+  def handle_cast(_late_msg, :completed), do: {:noreply, :completed}
+
+  def handle_cast({:report, %{name: name, attributes: attributes} = metric}, state) do
+    if nil_attributes?(attributes) do
+      NewRelic.log(
```

Comment:
https://github.com/newrelic/newrelic-telemetry-sdk-specs/blob/master/validation.md#sdk-validation

I thought I saw the SDK would check for attributes being null but I'm mistaken. I think we send it through for the backend to validate and generate Nr Integration errors. 

I don't think I would log a message here. It can be confusing since we are still sending it. I would remove this entirely.

Reply:
sounds good to me!